src/coreclr/jit/assertionprop.cpp

```diff
@@ -3992,6 +4200,15 @@ GenTree* Compiler::optAssertionPropGlobal_RelOp(ASSERT_VALARG_TP assertions, Gen
         return optAssertionProp_Update(newTree, tree, stmt);
     }
 
+    if (op2->IsCnsIntOrI() && tree->OperIs(GT_EQ, GT_LT, GT_LE, GT_GE, GT_GT))
```

Comment:
Does `GT_NE` not need need to be handled here?

Reply:
> Does `GT_NE` not need need to be handled here?

Heh, yeah, I was adding it separately - it's tricky because it's not representable via "{start, end}"